### DISCUSSION TITLE
Need help with jtl report

### DISCUSSION THREAD

**Discussion by leocharles003** [ludeknovy/jtl-reporter#60] [General]
Need step by step procedure on running jtl rpeort

**Comment by ludeknovy** [comment]
Hi @leocharles003.

Did you check the `Getting Started` in the docs: https://jtlreporter.site/docs/ ?

**Reply by leocharles003** [reply]
I shall want to know how it is working and parsing the jtl file in a detailed way

**Reply by ludeknovy** [reply]
You can find the code here: https://github.com/ludeknovy/jtl-reporter-be/blob/master/src/server/controllers/item/create-item-controller.ts

The whole process works like this
1. You upload the file
2. Data are parsed and streamed into mongodb
3. Aggregation pipelines are run
4. The aggregated data are saved into postgres
5. Uploaded file is deleted

About the parsing:
The actual CSV is parsing into object is done with 3rd party library, then the data are transformed (eg: string to numbers).
The code can be found here: https://github.com/ludeknovy/jtl-reporter-be/blob/73b0a4bed071c099e045a0a6187c28dbb6d0e6ee/src/server/data-stats/prepare-data.ts#L170

**Reply by leocharles003** [reply]
Thanks for sending the steps. Is there any UI to upload jtl file?

**Reply by ludeknovy** [reply]
You are welcome.

Yes, you can upload the file via UI, see: https://jtlreporter.site/docs/guides/manual-data-upload

**Reply by leocharles003** [reply]
ok, Is there any video tutorial for leveraging this utility ?. It will be really useful for many of them

**Reply by ludeknovy** [reply]
I dont think so. But feel free to create one ;)

**Reply by ludeknovy** [reply]
btw. there is a couple of blog post I wrote on how it can help you.

Is there anything specific you are after?
